inc/classes/class-offline-form.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * Offline form submission class.
+ *
+ * @package rt-pwa-extensions
+ */
+
+namespace RT\PWA\Inc;
+
+use \RT\PWA\Inc\Traits\Singleton;
+
+/**
+ * Class Offline_Form
+ */
+class Offline_Form {
+
+	use Singleton;
+
+	/**
+	 * @var array of error messages.
+	 */
+	private $error_messages;
+
+	/**
+	 * Construct method.
+	 */
+	protected function __construct() {
+
+		$this->error_messages = array(
+			'serverOffline' => esc_html__( 'The server appears to be down. Please try again later.', 'pwa-extension' ),
+			'error'         => esc_html__( 'Something prevented the page from being rendered. Please try again.', 'pwa-extension' ),
+			'form'          => esc_html__( 'Your form will be submitted once you are back online!', 'pwa-extension' ),
+		);
+
+		$this->setup_hooks();
+
+	}
+
+	/**
+	 * Setup actions/filters
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks() {
+
+		add_action( 'wp_front_service_worker', array( $this, 'offline_form_service_worker' ), 11 );
```

Comment:
Is there any specific reason to use 11 priority here. If there is we should specify in a comment